maitake-sync/src/wait_cell.rs

```diff
@@ -327,6 +327,46 @@ impl WaitCell {
         }
     }
 
+    /// Asynchronously poll the [`WaitCell`] until a condition occurs
+    ///
+    /// This can be used to implement a "wait loop", turning a "try" function
+    /// (e.g. "try_recv" or "try_send") into an asynchronous function (e.g.
+    /// "recv" or "send").
+    ///
+    /// Consider using [`Self::wait_for_value()`] if your function does return a value.
+    ///
+    /// * Returns `Ok(())` if the closure returns `true`.
+    /// * Returns `Err(Closed)` if the [`WaitCell`] is closed.
```

Comment:
tiny nit: elsewhere, the `maitake_sync` docs forrmat return value documentation as a `# Returns` heading --- e.g., in this module, see [here](https://github.com/hawkw/mycelium/blob/e456145e84ad6339752251318e6cf5d296b65d51/maitake-sync/src/wait_cell.rs#L114-L125):

```suggestion
    /// # Returns
    ///
    /// * [`Ok`]`(())` if the closure returns `true`.
    /// * [`Err`]`(`[`Closed`]`)` if the [`WaitCell`] is closed.
```